packages/synapse-react-client/src/components/EntityFinder/tree/EntityTree.tsx

```diff
@@ -413,38 +413,22 @@ export function EntityTree(props: EntityTreeProps) {
       <div className="Header">
         <div className="Browse">Browse:</div>
         <div onClick={e => e.stopPropagation()}>
-          <Dropdown>
-            <Dropdown.Toggle variant="gray-primary-500" id="dropdown-basic">
-              {scope}
-            </Dropdown.Toggle>
-            <Dropdown.Menu role="menu">
-              {Object.values(FinderScope).map(scopeOption => {
-                if (
+          <DropdownSelect
+            variant={'outlined'}
+            options={Object.values(FinderScope).filter(
+              scopeOption =>
+                !(
                   scopeOption === FinderScope.CURRENT_PROJECT &&
                   projectId == null
-                ) {
-                  return <React.Fragment key={scopeOption} />
-                }
-                return (
-                  <Dropdown.Item
-                    role="menuitem"
-                    key={scopeOption}
-                    onClick={e => {
-                      e.stopPropagation()
-                      if (scope !== scopeOption) {
-                        setScope(scopeOption)
-                        setCurrentContainer(
-                          getScopeOptionDefaultContainer(scopeOption),
-                        )
-                      }
-                    }}
-                  >
-                    {scopeOption}
-                  </Dropdown.Item>
-                )
-              })}
-            </Dropdown.Menu>
-          </Dropdown>
+                ),
+            )}
+            selectedIndex={Object.values(FinderScope).indexOf(scope)}
+            setSelectedIndex={index => {
+              const selectedScope = Object.values(FinderScope)[index]
+              setScope(selectedScope)
+              setCurrentContainer(getScopeOptionDefaultContainer(selectedScope))
+            }}
```

Comment:
Should this [still check](https://github.com/Sage-Bionetworks/synapse-web-monorepo/blob/506a12d9afd9ef45b99f94b2d423c50dcec24db1/packages/synapse-react-client/src/components/EntityFinder/tree/EntityTree.tsx#L434) whether the scope has changed (e.g. `scope !== selectedScope`) before updating the state with `setScope` and `setCurrentContainer`? I'm wondering if this check was added to prevent unnecessary re-renders, but @nickgros may know more.

Reply:
I believe the purpose of the check is to avoid resetting `currentContainer` when `scope` doesn't change.